<launch>
    <machine name="local" address="localhost" default="true"/>

    <!-- node name="Chisel" pkg="chisel_ros" type="ChiselNode" output="screen" launch-prefix="taskset -c 0,1" -->
    <node name="Chisel" pkg="chisel_ros" type="ChiselNode" output="screen" >
        <param name="chunk_size_x" value="8"/>  <!-- 8 -->
        <param name="chunk_size_y" value="8"/>
        <param name="chunk_size_z" value="8"/>
        <param name="truncation_scale" value="1.5"/>  <!-- 2 -->
        <param name="integration_weight" value="1.0"/>
        <param name="use_voxel_carving" value="true"/>
        <param name="carving_dist_m" value="0.01"/>  <!-- 0.10 -->
        <param name="voxel_resolution_m" value="0.15"/>  <!-- 0.1 for demo, 0.2 for bbgf 0.06 -->
        <param name="near_plane_dist" value="0.1"/>
        <param name="far_plane_dist" value="12.0"/>
        <param name="base_transform" value="/world"/>
        <param name="transform_name" value="/ref_frame"/>
        <param name="fusion_mode" value="DepthImage"/>

        <param name="three_camera_mode" value="false"/>
        <param name="calc_mesh" value="true"/>

        <param name="camera_model_file" value="$(find chisel_ros)/config/px_mei_camera_calib.yaml" />
        <param name="mask_file" value="$(find chisel_ros)/config/fisheye_mask.jpg" />

        <remap from="/odom_topic" to="/motion_stereo_left/odometry"/>
        <remap from="/depth_image" to="/motion_stereo_left/depth/image_raw"/>
        <remap from="/color_image" to="/motion_stereo_left/rgb/image_raw"/>
    </node>
</launch>
